### PR TITLE
Modern constraints polyfill in adapter.js (rebase)

### PR DESCRIPTION
**Pull request by jan-ivar**
A rebase of #22 .

### CODE REVIEW

File: adapter.js [hunk 1]
```diff
@@ -128,9 +167,64 @@ if (navigator.mozGetUserMedia) {
...
+      });
+      if (c.advanced) {
+        cc.optional = (cc.optional || []).concat(c.advanced);
+      }
+      return cc;
+    };
+    console.log('spec: ' + JSON.stringify(c));
+    c.audio = constraintsToChrome(c.audio);
+    c.video = constraintsToChrome(c.video);
+    console.log('chrm: ' + JSON.stringify(c));
```
Comment (alvestrand) [inline adapter.js]: Nit: please don't use abbreviations when we have plenty of room - "chrome:"

Comment (jan-ivar) [inline adapter.js]: Fwiw I did this for alignment reasons (not lack of room), e.g:

```
spec: {"video":{"width":1280,"height":720}}
ff37: {"video":{"advanced":[{"width":{"min":1280,"max":1280}},{"height":{"min":720,"max":720}}]}}
```

I can add whitespace I suppose.

**Comment by alvestrand** [comment]
Still missing tests. I want a log from Travis that shows that neither Chrome nor Firefox blow up when this polyfill of getUserMedia() gets called with a moderately-complex constraint set.

**Comment by jan-ivar** [comment]
@alvestrand I don't see any sign of tests running. Do I just wait?

**Comment by jan-ivar** [comment]
Side-note: I get a permission-dialog when I run this test in Chrome even though gUM always fails. That seems like a bug in Chrome .

**Comment by alvestrand** [comment]
On 05/28/2015 06:42 PM, jan-ivar wrote:
>
> @alvestrand <https://github.com/alvestrand> I don't see any sign of
> tests running. Do I just wait?
>

You rebase your PR on current head (which will get you the tests), write
tests into test/tests.js, and resubmit.

There won't be a test before you write one.

—
>
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/adapter/pull/29#issuecomment-106476413>.
>


-- 
Surveillance is pervasive. Go Dark.

**Comment by alvestrand** [comment]
@jan-ivar this is mysterious to me. The "fix console output" test got run by Travis; the latest one didn't.
I also got a "we can't automatically merge this pull request" - it may conflict with another merge.
Can you try rebasing again (git rebase master; git push --force)?

**Comment by jan-ivar** [comment]
@alvestrand pilot error on my part.

**Pull request closed without merging**